File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-04-02-preview/examples-js/ManagedClusterSnapshotsList.js
const { ContainerServiceClient } = require("@azure/arm-containerservice");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Gets a list of managed cluster snapshots in the specified subscription.
 *
 * @summary Gets a list of managed cluster snapshots in the specified subscription.
 * x-ms-original-file: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-04-02-preview/examples/ManagedClusterSnapshotsList.json
 */
async function listManagedClusterSnapshots() {
  const subscriptionId = process.env["CONTAINERSERVICE_SUBSCRIPTION_ID"] || "subid1";
  const credential = new DefaultAzureCredential();
  const client = new ContainerServiceClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.managedClusterSnapshots.list()) {
    resArray.push(item);
  }
  console.log(resArray);
}

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2026-03-02-preview/examples-java/ManagedClusterSnapshotsList.java

/**
 * Samples for ManagedClusterSnapshots List.
 */
public final class Main {
    /*
     * x-ms-original-file: 2026-03-02-preview/ManagedClusterSnapshotsList.json
     */
    /**
     * Sample code: List Managed Cluster Snapshots.
     * 
     * @param manager Entry point to ContainerServiceManager.
     */
    public static void
        listManagedClusterSnapshots(com.azure.resourcemanager.containerservice.ContainerServiceManager manager) {
        manager.serviceClient().getManagedClusterSnapshots().list(com.azure.core.util.Context.NONE);
    }
}
